code/modules/mob/mob.dm

```diff
@@ -1455,6 +1455,8 @@
 
 /mob/proc/update_equipment_speed_mods()
 	var/speedies = equipped_speed_mods()
+	if(HAS_TRAIT(src, TRAIT_SETTLER) && speedies > 0) //if our movespeed mod is in the negatives, we don't modify it since that's a benefit
```

Comment:
rearrange this so the `speedies` check comes first for effective shortcircuiting because a simple `>` numerical check is way faster than the `HAS_TRAIT()` macro. no sense doing the work on the has_trait stuff if we were going to fail the second check anyhow

Reply:
can do